econml/solutions/causal_analysis/_causal_analysis.py

```diff
@@ -104,60 +105,67 @@ def _get_metadata_causal_insights_keys():
             _CausalInsightsConstants.ViewKey]
 
 
-def _first_stage_reg(X, y, *, automl=True):
+def _first_stage_reg(X, y, *, automl=True, random_state=None, verbose=0):
     if automl:
-        model = GridSearchCVList([make_pipeline(StandardScaler(), LassoCV()),
+        model = GridSearchCVList([make_pipeline(StandardScaler(), LassoCV(random_state=random_state)),
                                   RandomForestRegressor(
-                                      n_estimators=100, random_state=123, min_samples_leaf=10),
-                                  lgb.LGBMRegressor(num_leaves=32)],
+                                      n_estimators=100, random_state=random_state, min_samples_leaf=10),
+                                  lgb.LGBMRegressor(num_leaves=32, random_state=random_state)],
                                  param_grid_list=[{},
                                                   {'min_weight_fraction_leaf':
                                                       [.001, .01, .1]},
                                                   {'learning_rate': [0.1, 0.3], 'max_depth': [3, 5]}],
                                  cv=2,
```

Comment:
why do we have cv=2 in the regression and cv=5 in the classification?

Reply:
Also it seems that neg_mse is a very cryptic score and we might want to switch to r2score for regression and log loss for classification .

Reply:
or even simpler, switch to r2score at least for both, so that the score is normalized in [0, 1]

Reply:
Switched to `cv=3` for both, and switched regression to r2score; not sure r2score makes sense for a classifier so didn't change that one now, but can revisit if you feel strongly.